src/Kernel.sol

```diff
@@ -35,7 +35,12 @@ struct Instruction {
     address target;
 }
 
-// ######################## ~ CONTRACT TYPES ~ ########################
+struct RequestPermissions {
```

Comment:
Would rename to `Permissions` just to be concise. `requestPermissions` is already used by the function.